plugins/providers/gcs/config.go

```diff
@@ -0,0 +1,194 @@
+package gcs
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/mitchellh/mapstructure"
+	"github.com/odpf/guardian/domain"
+)
+
+const (
+	BucketRoleAdmin         = "roles/storage.admin"
+	BucketRoleOwner         = "roles/storage.legacyBucketOwner "
+	BucketRoleReader        = "roles/storage.legacyBucketReader"
+	BucketRoleWriter        = "roles/storage.legacyBucketWriter"
+	BucketRoleObjectOwner   = "roles/storage.legacyObjectOwner"
+	BucketRoleObjectReader  = "roles/storage.legacyObjectReader"
+	BucketRoleObjectAdmin   = "roles/storage.objectAdmin"
+	BucketRoleObjectCreator = "roles/storage.objectCreator"
+	BucketRoleObjectViewer  = "roles/storage.objectViewer"
+
+	AccountTypeUser           = "user"
+	AccountTypeServiceAccount = "serviceAccount"
+	AccountTypeGroup          = "group"
+	AccountTypeDomain         = "domain"
+)
+
+type Config struct {
+	ProviderConfig *domain.ProviderConfig
+	valid          bool
+
+	crypto    domain.Crypto
+	validator *validator.Validate
+}
+
+func (c *Config) EncryptCredentials() error {
+	if err := c.parseAndValidate(); err != nil {
+		return err
+	}
```

Comment:
this won't be needed if we sure the credential is always valid before this `EncrypCredentials` being called right?

Reply:
@rahmatrhd should we update all the other providers with this, I checked this function is being called only once in all the providers. Apart form this the valid bool will also be removed since it was being used to cache for this call only

Reply:
@Chief-Rishab we can let the other providers problem to another PR. Each provider might have different behaviour so IMO we can try to keep this provider simpler.

Reply:
sure, will do it for gcs then

Reply:
done in https://github.com/odpf/guardian/pull/227/commits/b70e3ec4aa8e55ff4938dd43d641c1ce007e8bc3